src/Cli/dotnet/commands/dotnet-sln/migrate/SlnMigrateCommand.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools;
+using Microsoft.VisualStudio.SolutionPersistence;
+using Microsoft.VisualStudio.SolutionPersistence.Model;
+using Microsoft.VisualStudio.SolutionPersistence.Serializer;
+using LocalizableStrings = Microsoft.DotNet.Tools.Sln.LocalizableStrings;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal class SlnMigrateCommand : CommandBase
+    {
+        private readonly string _slnFileFullPath;
+        private readonly IReporter _reporter;
+        public SlnMigrateCommand(
+            ParseResult parseResult,
+            IReporter reporter = null)
+            : base(parseResult)
+        {
+            _slnFileFullPath = Path.GetFullPath(parseResult.GetValue(SlnCommandParser.SlnArgument));
+            _reporter = reporter ?? Reporter.Output;
+        }
+
+        public override int Execute()
+        {
+            if (!File.Exists(_slnFileFullPath) || !Path.GetExtension(_slnFileFullPath).EndsWith("sln"))
+            {
+                throw new GracefulException(CommonLocalizableStrings.CouldNotFindSolutionIn, _slnFileFullPath);
+            }
+            string slnxFileFullPath = Path.ChangeExtension(_slnFileFullPath, "slnx");
+            Task task = ConvertToSlnxAsync(_slnFileFullPath, slnxFileFullPath, CancellationToken.None);
```

Comment:
I believe you will want to call `RunSynchronously()` on this `Task`. But I haven't used Tasks in eons, so I would check with this.

Reply:
After chatting with @rainersigwald, I believe you can call .Wait() on this task in a try catch, if it succeeds then it will continue on and you can return 0, otherwise in the catch you can catch an error and go from there.